synapse/handlers/sync.py

```diff
@@ -1413,10 +1414,10 @@ async def generate_sync_result(
     async def _generate_sync_entry_for_device_list(
         self,
         sync_result_builder: "SyncResultBuilder",
-        newly_joined_rooms: Set[str],
-        newly_joined_or_invited_or_knocked_users: Set[str],
-        newly_left_rooms: Set[str],
-        newly_left_users: Set[str],
+        newly_joined_rooms: AbstractSet[str],
```

Comment:
Not sure it matters, but this is [now deprecated as of 3.9](https://docs.python.org/3/library/typing.html#typing.AbstractSet)?

Reply:
That's a good point.

Annoyingly, `collections.abc.Set` doesn't support `[]` until Python 3.9, so using it now would prove troublesome for development. I suppose once 3.8 hits end of life, we'll want to update all our `typing` imports to use `collections.abc` instead.

```py
$ docker run -it python:3.8-alpine
[...]
>>> import collections.abc
>>> collections.abc.Set[int]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'ABCMeta' object is not subscriptable
```
```py
$ docker run -it python:3.9-alpine
[...]
>>> import collections.abc
>>> collections.abc.Set[int]
collections.abc.Set[int]
```

Reply:
That was what I assumed... so I guess this is fine for now!